android/java/org/chromium/chrome/browser/crypto_wallet/activities/AddAccountActivity.java

```diff
@@ -87,93 +89,22 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
-                                importAccountPasswordText.getText().toString(),
-                                mPrivateKeyControl.getText().toString(), (result, address) -> {
-                                    if (result) {
-                                        setResult(Activity.RESULT_OK);
-                                        Utils.clearClipboard(
-                                                mPrivateKeyControl.getText().toString(), 0);
-                                        Utils.clearClipboard(
-                                                importAccountPasswordText.getText().toString(), 0);
-                                        finish();
-                                    } else {
-                                        mAddAccountText.setError(getString(
-                                                R.string.wallet_failed_to_import_account));
-                                    }
-                                });
-                    } else {
-                        mKeyringService.importAccount(mAddAccountText.getText().toString(),
-                                mPrivateKeyControl.getText().toString().trim(),
-                                mCryptoAccountTypeInfo.getCoinType(), (result, address) -> {
-                                    if (result) {
-                                        setResult(Activity.RESULT_OK);
-                                        Utils.clearClipboard(
-                                                mPrivateKeyControl.getText().toString(), 0);
-                                        finish();
-                                    } else {
-                                        mAddAccountText.setError(getString(
-                                                R.string.wallet_failed_to_import_account));
-                                    }
-                                });
-                    }
-                } else {
-                    mWalletModel.getKeyringModel().addAccount(mAddAccountText.getText().toString(),
-                            mCryptoAccountTypeInfo.getCoinType(), result -> {
-                                if (result) {
-                                    setResult(Activity.RESULT_OK);
-                                    finish();
-                                } else {
-                                    mAddAccountText.setError(
-                                            getString(R.string.account_name_empty_error));
-                                }
-                            });
-                }
+            if (mKeyringService == null) return;
+            @CoinType.EnumType
+            int coinType = mCryptoAccountTypeInfo.getCoinType();
+            if (mIsUpdate) {
+                updateAccountName(coinType);
+            } else if (!TextUtils.isEmpty(mPrivateKeyControl.getText().toString())) {
+                importAccount(coinType);
+            } else {
+                addAccount(coinType);
```

Comment:
I've refactored the code inside the listener as it was quite hard to read. I hope it makes it easier to follow now.
The logic was left untouched, except for using the new Filecoin methods when `coinType == FIL`.